### ISSUE TITLE
skip SSL error webview_flutter

### ISSUE THREAD

**Issue by mrpixel04** [flutter/flutter#36925]
Hi there..

I'm using the Flutter plugins 'webview_flutter' to load certain url and it works perfectly.. with http or without https .... but there is certains url comes with https was detected having not a valid ssl caused my flutter apps to be blank...

So my questions is...is that possible I can ignored those SSL errors and just proceed with the url using this 'webview_flutter' plugins..anyone got luck with it...or maybe point me in the right direction....

Thanks in advanced

**Comment by timsneath** [comment]
Per [#2285](https://github.com/flutter/plugins/pull/2285), this is by design: 

> By default, Flutter plugins should expect or enforce a secure environment, therefore we can not process this PR as-is. However, I can see a use case, especially during local development when it's just may not be feasible to have a properly signed https connection.

**Comment by stuartmorgan** [comment]
The comment quoted above was about a PR that simply turned off all SSL handling; allowing an *option* to bypass checks is a valid request.

**Comment by sherpya** [comment]
what about a callback to onReceivedSslError and/or use flutter's SecurityContext, so one can add his own certs and manually validate certificates like in dart:io?

**Comment by devmanmath** [comment]
Any update on how to add ignoreSSL error property to webView in flutter?

**Comment by chaudharydeepanshu** [comment]
This is really required for:-

1. temporary bypassing ssl errors during development 
2. **non-production** apps in which the developer doesn't have access to the website causing ssl errors but wants to specifically trust it.

From **flutter_webview_plugin** which has option for ignoring SSL Errors:

> We are working closely with the Flutter Team to **integrate all the Community Plugin features in the Official WebView Plugin**. We will try our best to resolve PRs and Bugfixes, but our priority right now is to merge our two code-bases. **Once the merge is complete we will deprecate the Community Plugin in favor of the Official one**.

So, I think it is not out of scope for "webview_flutter" as many applications created using "flutter_webview_plugin" will have to migrate to "webview_flutter" once "flutter_webview_plugin" deprecates to continue app development and maintain stability.

**Comment by mdmota** [comment]
until version 2.0.14 they had the android and ios folders. As of version 2.1.0 there are no more folders.

https://github.com/flutter/plugins/tree/webview_flutter-v2.0.14/packages/webview_flutter/webview_flutter

I need to customize webview client, how to do this in the latest version?

**Comment by mdmota** [comment]
until version 2.0.14 they had the android and ios folders. As of version 2.1.0 there are no more folders.

https://github.com/flutter/plugins/tree/webview_flutter-v2.0.14/packages/webview_flutter/webview_flutter

**Comment by mdmota** [comment]
Any update?

**Comment by Jalil-Irfan** [comment]
Any updates , please

**Comment by lokeb** [comment]
Hi. Why is this not working with custom installed CA certificate on the device? The default webview example on the emulator works fine with the installed certificate but flutter_webview still throws handshake error.

**Comment by SeltiOrg** [comment]
> ```dart
>  //Do some checks here to decide if CANCELS or PROCEEDS
> ```

 `//Do some checks here to decide if CANCELS or PROCEEDS`
Does anyone know how to check a self-signed certificate here?

**Comment by SeltiOrg** [comment]
> I had the same error described in this issue and opted to find another WebView implementation which allows me to manage certificate validation. I finally ended up using [flutter_inappwebview](https://pub.dev/packages/flutter_inappwebview) with a code like this:
> 
> ```dart
> InAppWebView(
>       initialUrlRequest: URLRequest(url: Uri.parse(myCustomUrl)),
>       onReceivedServerTrustAuthRequest: (controller, challenge) async {
>         //Do some checks here to decide if CANCELS or PROCEEDS
>         return ServerTrustAuthResponse(action: ServerTrustAuthResponseAction.PROCEED);
>       },
>     )
> ```
> 
> I hope it helps.

`//Do some checks here to decide if CANCELS or PROCEEDS`

Does anyone know how to check a self-signed certificate here?

**Comment by hainey** [comment]
Moving from flutter_webview_plugin to webview_flutter are painful, 
Without ignoresslerrors feature, I have to copy the latest webview_flutter plugin source to my own project and do  customization
:-(

**Comment by chitgoks** [comment]
following this. this request is still valid because other features like url_launcher uses webview_flutter and it does not load a url with an invalid ssl certificate

**Comment by stuartmorgan** [comment]
> other features like url_launcher uses webview_flutter

This is not accurate; `url_launcher` does not use this plugin.

**Comment by chitgoks** [comment]
> > other features like url_launcher uses webview_flutter
> 
> This is not accurate; `url_launcher` does not use this plugin.

doesnt url launcher have an option forceWebView?

**Comment by lyf571321556** [comment]
any update?

**Comment by andrixnet** [comment]
IMO the best policy would be just like browsers: 
- expose failed validity reason
- show certificate details
- allow user to manually validate a bypass

**Comment by sed1ka** [comment]
It's been 5 years, but this feature still hasn't been implemented. Wanted to try to use another library but `webview_flutter` is official. Meanwhile, `flutter_inappwebview` has lots of bugs. ༼ ༎ຶ ෴ ༎ຶ༽

**Comment by sed1ka** [comment]
I always getting
[ERROR:ssl_client_socket_impl.cc(996)] handshake failed; returned -1, SSL error code 1, net_error -202

Even though the URL I opened has a valid SSL. The `onWebResourceError` also doesn't trigger at all when the SSL error appears.

**Comment by haizadvnet** [comment]
Hi, any forked versions of webview_flutter that can bypass the ssl checks? I am fully aware on the important of having SSL certificates, but I need this for local development only.

**Comment by devmarwat** [comment]
> I always getting [ERROR:ssl_client_socket_impl.cc(996)] handshake failed; returned -1, SSL error code 1, net_error -202
> 
> Even though the URL I opened has a valid SSL. The `onWebResourceError` also doesn't trigger at all when the SSL error appears.

Facing the Same Issue. [ERROR:ssl_client_socket_impl.cc(882)] handshake failed; returned -1, SSL error code 1, net_error -202
